pyEpiabm/pyEpiabm/sweep/spatial_sweep.py

```diff
@@ -42,52 +45,117 @@ def __call__(self, time: float):
+                # Chooses cells based on a cumulative transmission array
+                # one and a time a tests each infection event.
+                while number_to_infect > 0:
+                    # Weighting for cell choice in Covidsim uses cum_trans and
+                    # invCDF arrays, but really can't see how these are
+                    # initialised. Have used the number of susceptible for now.
+                    weights = [cell2.compartment_counter.retrieve()
+                               [InfectionStatus.Susceptible]
+                               for cell2 in possible_infectee_cells]
+                    infectee_cell = random.choices(possible_infectee_cells,
+                                                   weights=weights, k=1)[0]
+                    # Sample at random from the infectee cell to find
+                    # an infectee
+                    infectee = random.sample(infectee_cell.persons, 1)[0]
+                    infection_distance = DistanceFunctions.dist(
+                     cell.location, infectee_cell.location) / Parameters.\
+                        instance().infection_radius
+                    if (infection_distance < random.random()):
+                        # Covidsim rejects the infection event if the distance
+                        # between infector/infectee is too large.
+                        self.do_infection_event(infector, infectee, timestep)
+                        number_to_infect -= 1
+
+            else:
+                # Chooses a list of cells (with replacement) for each infection
+                # event to occur in. Specifically inter-cell infections
+                # so can't be the same cell.
+                distance_weights = []
+                # Possibly don't want this to be unfeasibly large as still
+                # have a valid population with clumps of cells
+                for cell2 in possible_infectee_cells:
+                    try:
+                        distance_weights.append(1/DistanceFunctions.dist(
+                                    cell.location, cell2.location))
+                    except ZeroDivisionError:
+                        # If cells are on top of each other use nan placeholder
+                        distance_weights.append(np.nan)
+                # Cells on top of each currently have a distance weight equal
+                # to the maximum of all other weights.
+                # Possibly want to do twice this.
+                number_of_nans = sum(np.isnan(distance_weights))
+                if number_of_nans == len(distance_weights):
+                    distance_weights = [1 for _ in distance_weights]
+                elif number_of_nans > 0:
+                    max_weight = np.nanmax(distance_weights)
+                    distance_weights = [max_weight if np.isnan(weight)
+                                        else weight for weight in
+                                        distance_weights]
```

Comment:
```suggestion
                    distance_weights = np.nan_to_num(distance_weights, nan=max_weight)
```

Much cleaner (&faster), I have not ran tests on this (and coding without autocomplete etc is surprisingly hard) but should work